### ISSUE TITLE
Crash in "View all sensors" on the new M3 Macbook Pro. Also no labeling for sensors.

### ISSUE THREAD

**Issue by varenc** [macmade/Hot#81]
Hot mostly works, but when I try to open up the View All Sensors view and scroll around it crashes. If I don't scroll it doesn't crash, but as soon as I scroll a little bit it crashes.

Attached is the macOS crash report.
[hot_crash_M3_mbp.txt](https://github.com/macmade/Hot/files/13629472/hot_crash_M3_mbp.txt)

The crash seems to be caused here:
```
  0   CoreFoundation                      0x0000000189706800 __exceptionPreprocess + 176
  1   libobjc.A.dylib                     0x00000001891fdeb4 objc_exception_throw + 60
  2   CoreFoundation                      0x00000001897066f0 +[NSException exceptionWithName:reason:userInfo:] + 0
  3   AppKit                              0x000000018cf8c88c -[NSBezierPath lineToPoint:] + 72
  4   Hot                                 0x0000000104fb9c8c Hot + 171148
  5   Hot                                 0x0000000104fb8b9c Hot + 166812
  6   Hot                                 0x0000000104fb8c4c Hot + 166988
  7   AppKit                              0x000000018cf83b20 _NSViewDrawRect + 124
```

While I'm here I'll mention that Hot also has poor labeling for the various M3 Macbook Pro sensors.
<img width="471" alt="image" src="https://github.com/macmade/Hot/assets/167585/81aadc6b-cccd-4a09-868b-e079d3f0e617">

The CLI tool `iSMC` also struggles with labeling the sensors, though it does a better job at labeling at least some of them. Also attached is the iSMC output where you can see that at least a few of the sensors get helpful labels.
[iSMC_mbp.json](https://github.com/macmade/Hot/files/13629476/iSMC_mbp.json)
[iSMC_mbp.txt](https://github.com/macmade/Hot/files/13629477/iSMC_mbp.txt)

**Comment by katomic** [comment]
On a M1 with MacOs 141.2, I can see these exact same issues.

**Comment by frasersdev** [comment]
Same crash & lack of sensor ID on a M2 Max Studio running 14.1.2 

iSMC identified these:

```
                     Current                     
 DESCRIPTION                KEY   VALUE     TYPE 
 Mainboard S0 Rail (DC In)  ID0R     0.8 A  flt  
 PMU ibuck3                        15.85 A  hid  
 PMU ibuck8                        16.03 A  hid  
 PMU ibuck9                        16.00 A  hid  
 PMU ildo1                         15.78 A  hid  
 PMU ildo2                         15.52 A  hid  
 PMU ildo5                         15.53 A  hid  

                    Fans                   
 DESCRIPTION          KEY   VALUE     TYPE 
 Fan 1 Current Speed  F0Ac  1000 rpm  flt  
 Fan 1 Maximum Speed  F0Mx  3500 rpm  flt  
 Fan 1 Minimal Speed  F0Mn  1000 rpm  flt  
 Fan 1 Target Speed   F0Tg  1000 rpm  flt  
 Fan 2 Current Speed  F1Ac   995 rpm  flt  
 Fan 2 Maximum Speed  F1Mx  3500 rpm  flt  
 Fan 2 Minimal Speed  F1Mn  1000 rpm  flt  
 Fan 2 Target Speed   F1Tg  1000 rpm  flt  
 Fan Count            FNum         2  ui8  

                  Power                  
 DESCRIPTION        KEY   VALUE     TYPE 
 Average CPU 1V5    PD1R     0.0 W  flt  
 DC In              PDTR     9.7 W  flt  
 Heatpipe           PHPC     4.6 W  flt  
 Mainboard S0 Rail  PD0R     9.7 W  flt  
 System Total       PSTR     9.7 W  flt  

                           Temperature                           
 DESCRIPTION                                KEY   VALUE     TYPE 
 CPU Efficiency Core 1                      Tp09   39.9 °C  flt  
 CPU Performance Core 1                     Tp01   40.6 °C  flt  
 CPU Performance Core 2                     Tp05   37.7 °C  flt  
 CPU Performance Core 3                     Tp0D   37.5 °C  flt  
 CPU Performance Core 7                     Tp0X   39.7 °C  flt  
 CPU Performance Core 8                     Tp0b   37.6 °C  flt  
 Drive 0 OOBv3 Absolute Raw A               TH0a   30.5 °C  flt  
 Drive 0 OOBv3 Absolute Raw B               TH0b   30.6 °C  flt  
 Drive 0 OOBv3 Max                          TH0x   30.6 °C  flt  
 Heat Spreader Proximity                    Th0P   32.6 °C  flt  
 Heatpipe 1                                 Th0H   32.7 °C  flt  
 NAND                                       TH0x   30.6 °C  flt  
 NAND CH0 temp                                    30.00 °C  hid  
 PMU TP0s                                         35.80 °C  hid  
 PMU TP1g                                         35.18 °C  hid  
 PMU TP1s                                         35.62 °C  hid  
 PMU TP2g                                         35.53 °C  hid  
 PMU TP2s                                         35.80 °C  hid  
 PMU TP3g                                         35.09 °C  hid  
 PMU tcal                                         51.85 °C  hid  
 PMU tdev1                                        34.00 °C  hid  
 PMU tdev2                                        33.71 °C  hid  
 PMU tdev3                                        33.76 °C  hid  
 PMU tdev4                                        34.10 °C  hid  
 PMU tdev5                                        32.94 °C  hid  
 PMU tdev6                                        35.93 °C  hid  
 PMU tdev7                                        32.70 °C  hid  
 PMU tdev8                                        33.23 °C  hid  
 PMU tdie0                                        35.44 °C  hid  
 PMU tdie1                                        34.73 °C  hid  
 PMU tdie2                                        35.26 °C  hid  
 PMU tdie3                                        35.62 °C  hid  
 PMU tdie4                                        35.80 °C  hid  
 PMU tdie5                                        35.71 °C  hid  
 PMU tdie6                                        35.44 °C  hid  
 PMU tdie7                                        35.44 °C  hid  
 PMU tdie8                                        35.89 °C  hid  
 PMU tdie9                                        36.15 °C  hid  
 PMU tdie10                                       35.35 °C  hid  
 Power Supply 1 Alt                         Tp0C   32.9 °C  flt  
 Power Supply 2 Alt                         Tp1C   45.2 °C  flt  
 Power Supply T2 Secondary Heatsink Cooked  Tp2H   39.2 °C  flt  
 Speaker Proximity                          TS0P   33.6 °C  flt  
 Speaker Proximity 2                        TS1P   32.7 °C  flt  

              Voltage              
 DESCRIPTION  KEY   VALUE     TYPE 
 DC In        VD0R    12.1 V  flt  
 PMU VP0s            16.29 V  hid  
 PMU vbuck0          16.21 V  hid  
 PMU vbuck1          16.21 V  hid  
 PMU vbuck3          16.36 V  hid  
 PMU vbuck4          16.26 V  hid  
 PMU vbuck5          16.17 V  hid  
 PMU vbuck6          16.27 V  hid  
 PMU vbuck7          16.16 V  hid  
 PMU vbuck8          16.29 V  hid  
 PMU vbuck9          16.21 V  hid  
 PMU vbuck10         16.15 V  hid  
 PMU vldo1           16.47 V  hid  
 PMU vldo2           16.36 V  hid  
 PMU vldo5           16.20 V  hid  
```

**Comment by WWWFWWW** [comment]
Same issue on M1. The associated standalone Sensors crashes the same way as well.

Also, both the View all Sensors and the Sensors.app are pretty CPU intensive (nearly 90% usage), instantly raising the temperatures by a lot periodically:
![tmp](https://github.com/macmade/Hot/assets/30300299/232472c6-592a-47c3-a8cc-8af9013e0623)

No idea if that's just the normal polling and expected, display or some other issues.

**Comment by intrak** [comment]
I have the same problem.
MBP - M1 Pro,
macOS 14.5 (23F79)

**Comment by macmade** [comment]
Fixed in 1.9.2: https://github.com/macmade/Hot/releases/tag/1.9.2
